lib/parser.js

```diff
@@ -180,6 +184,37 @@ function skipIfBlock(chunk, state) {
   return curlyBraceCount === 0;
 }
 
+function skipFunctionDefinition(chunk, state) {
+  var start = state.index;
+  var parenthesisNest = 1;
+  var character = chunk[++state.index];
+  while (character !== undefined && parenthesisNest) {
+    if (character === CHAR_LEFT_PARENTHESIS) {
+      parenthesisNest++;
+    } else if (character === CHAR_RIGHT_PARENTHESIS) {
+      parenthesisNest--;
+    }
+
+    character = chunk[++state.index];
+  }
+
+  while (character && character !== CHAR_BLOCK_START) character = chunk[++state.index];
```

Comment:
Minor: Could you wrap the assignment within a pair of curly braces?  Apart from that everything looks and works great!